src/regex/Regex.java

```diff
@@ -1,46 +1,110 @@
+        final Matcher sleepyMatcher = sleepyPattern.matcher(sleepy);
 
         // 4.b combine the array that you just created into a string
-        String not_sleepy; //punctuation marks will be here
+        System.out.println("Problem 4b Start");
+        System.out.println(
+                "=========================================================================================================");
+        String not_sleepy = sleepyMatcher.replaceAll(""); //punctuation marks will be here
+        System.out.println(not_sleepy);
+        System.out.println("Problem 4b End");
+        System.out.println(
+                "=========================================================================================================");
 
         //4.c remove the punctuation marks from the notSleepy string.
+        System.out.println("Problem 4c Start");
+        System.out.println(
+                "=========================================================================================================");
         String notSleepy; //no punctuation marks should be here.
+        final String punctuationRemove = "(sleepy)|[.',]";
+        final Pattern punctuationRemovePattern = Pattern.compile(punctuationRemove,
+                Pattern.MULTILINE | Pattern.CASE_INSENSITIVE);
+        final Matcher punctuationRemoveMatcher = punctuationRemovePattern.matcher(sleepy);
+        notSleepy = punctuationRemoveMatcher.replaceAll("");
+        System.out.println(notSleepy);
+        System.out.println("Problem 4c End");
+        System.out.println(
+                "=========================================================================================================");
 
         //4.d Now replace all the occurences of 'sleepy' with the word 'happy'.  Call the new string happy.  
-
-        String happy;
+        System.out.println("Problem 4d Start");
+        System.out.println(
+                "=========================================================================================================");
+        final String subst2 = "happy";
+        final String happy = sleepyMatcher.replaceAll(subst2);
+        System.out.println(happy);
+        System.out.println("Problem 4d End");
+        System.out.println(
+                "=========================================================================================================");
 
         //BONUS : 
         //5. You are looking for unicode arrow symbols in a string.  https://jrgraphix.net/r/Unicode/2190-21FF is a selection of unicode arrow symbols to aid you in your search.  Match all the codes that are arrows, and then print them out to the console.  They should be printing out as the arrow images.
 
-        String[] arrows = {"\u21FD", "\u26F7", "\u21FF", "\u21EF","\u21EC", "\u26F9","\u26FD","\u26D4","\u26A5","\u21FD","\u2190", "\u26A1","\u21BA","\u2196","\u2603","\u21FD"};
```

Comment:
skipped extra credit (but I do like that you auto-formatted the code, to keep it more-readable)